### PR TITLE
[REEF-951] Allow empty string as default for a named parameter

### PR DESCRIPTION
**Pull request by jwang98052**
Similar change as in REEF-950
Redefine internally used default and allow the user to use empty string as default
Add test cases
Fix default setting for serviceSet

JIRA: [REEF-951](https://issues.apache.org/jira/browse/REEF-951)

This closes #

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Tang.Tests/Tang/TestTang.cs [hunk 1]
```diff
@@ -831,8 +840,36 @@ private ABtaker([Parameter(typeof (AImplName))] INamedImplA a, [Parameter(typeof
...
+    public class EmptyStringAsDefaultParam : Name<string>
+    {
+    }
+
+    public class EmptyStringAsDefaultParamConf : ConfigurationModuleBuilder
+    {
+        public static readonly OptionalParameter<string> OptionalString = new OptionalParameter<string>();
+
+        public static ConfigurationModule ConfigurationModule = new EmptyStringAsDefaultParamConf()
+       .BindNamedParameter(GenericType<EmptyStringAsDefaultParam>.Class, OptionalString)
```
Comment (dongjoon-hyun) [inline lang/cs/Org.Apache.REEF.Tang.Tests/Tang/TestTang.cs]: Indentation?

File: lang/cs/Org.Apache.REEF.Tang.Tests/Tang/TestTang.cs [hunk 2]
```diff
@@ -831,8 +840,36 @@ private ABtaker([Parameter(typeof (AImplName))] INamedImplA a, [Parameter(typeof
...
+    public class StringWithoutDefaultParam : Name<string>
+    {
+    }
+
+    public class StringWithoutDefaultParamConf : ConfigurationModuleBuilder
+    {
+        public static readonly OptionalParameter<string> OptionalString = new OptionalParameter<string>();
+
+        public static ConfigurationModule ConfigurationModule = new StringWithoutDefaultParamConf()
+       .BindNamedParameter(GenericType<StringWithoutDefaultParam>.Class, OptionalString)
```
Comment (dongjoon-hyun) [inline lang/cs/Org.Apache.REEF.Tang.Tests/Tang/TestTang.cs]: Indentation?

**Comment by dongjoon-hyun** [comment]
It looks good to me. I will test and merge this.

**Comment by jwang98052** [comment]
@dongjoon-hyun Thanks!

**Pull request closed without merging**